tests/store/test_tag_filters.py

```diff
@@ -0,0 +1,71 @@
+from housekeeper.store import Store
+from housekeeper.store.models import Tag
+from housekeeper.store.tag_filters import filter_tag_by_name, TagFilter, apply_tag_filter
+from sqlalchemy.orm import Query
+from typing import List
+
+
+def test_filter_tag_by_name_returns_correct_tag(populated_store: Store, sample_tag_name: str):
+    """Test that fetching a tag by name returns a tag with the expected name."""
+
+    # GIVEN a store with the desired tag
+    all_tags: Query = populated_store._get_tag_query()
+    all_tag_names: List[str] = [t.name for t in all_tags.all()]
+    assert sample_tag_name in all_tag_names
+
+    # WHEN retrieving the tag by name
+    tag: Tag = filter_tag_by_name(
+        tags=all_tags,
+        tag_name=sample_tag_name,
+    ).first()
+
+    # THEN a tag should be returned
+    assert isinstance(tag, Tag)
```

Comment:
I think we should be mindful that when we are testing filters to explicitly test what they are supposed to return.

```suggestion
        # WHEN retrieving the tag by name
    tag: Tag = filter_tag_by_name(
        tags=all_tags,
        tag_name=sample_tag_name,
    )

    # THEN a tag should be returned
    assert isinstance(tag, Query)
```

Reply:
done!